src/components/number-picker/number-picker.component.ts

```diff
@@ -15,7 +15,7 @@ export const NUMBER_PICKER_VALUE_ACCESSOR: any = {
     templateUrl: './number-picker.component.html',
     providers: [NUMBER_PICKER_VALUE_ACCESSOR],
     host: {
-        '[class.has-error]': '!isValid()'
+        '[class.has-error]': '!_valid && !disabled'
```

Comment:
With the new `.ng-invalid` style, can this now be removed?

Reply:
I don't think we can as we still support the `[(value)]` input/output, and `ng-invalid` will only be present when either `ngModel` or `ReactiveForms` are being used.

Reply:
Are there any recommendations or existing Angular controls which work this way? I feel like forcing validation appearance may continue to cause problems. Like we previously overlooked the disabled case, which is now baked into the component, but consumers may have other complex validation rules within their form layout in which case we would only want the .ng-invalid rule to apply.

Reply:
I can't think of any other components of the top of my head that work this way. I agree, this was something that was overlooked at the time, and I would rather not have the component handle any validation itself but use Angular validation only.

Currently we have a `[valid]` input which a user can set to `true` if they want to show the red outline and we also have `[min]` and `[max]` inputs which are also used to validate the input. This isn't ideal as `ngModel` uses the `min` and `max` HTML attributes to also perform validation. 

I'm not sure how we can alter this without a breaking change as we have an input that can be specifically used to set the red outline,

Reply:
That's true. I actually think we ought to deprecate the `valid` input and advise to either use reactive forms, or apply `has-error` to the parent element (the Bootstrap way). Not sure what to do about min and max but with 2.0 coming up we do have an opportunity to make breaking changes.

In terms of current behavior, one idea is to change the name of the class from `has-error` to something specific like `ux-number-picker-invalid`. We can then make a style rule like

````css
ux-number-picker:not([formControlName]).ux-number-picker-invalid {
    border-color: red;
}
````

or something along those lines so that it only applies when the number picker is not part of a reactive form.

Reply:
Yeah, I agree, with 2.0 looming now is the time to deprecate things like this. Yeah, that makes sense, I think renaming the class is the best way rather than having the component update a class on something outside of it's template.